jetty-server/src/main/java/org/eclipse/jetty/server/ContentProducer.java

```diff
@@ -148,5 +150,18 @@
      * @return true if the thread has to be rescheduled, false otherwise.
      */
     boolean onContentProducible();
-}
 
+    /**
+     * This exception used to report when there is some unconsumed content left at the end of a request's processing.
+     * Suppressed exceptions are disabled, meaning calling {@link #addSuppressed(Throwable)} has no effect.
+     */
+    class UnconsumedContentException extends Exception
+    {
+        private static final Logger LOG = LoggerFactory.getLogger(UnconsumedContentException.class);
+
+        public UnconsumedContentException()
+        {
+            super("Unconsumed content", null, false, LOG.isDebugEnabled());
```

Comment:
I think this constructor is key!
Seeing that this constructor allows us to disable suppressed exceptions, there is no real reason for us to avoid static Throwables used as sentinels.

I'm not keen on using the LOG to hide stack traces.  I think if it is worth creating an exception of the fly, it is worth generating a stack trace.  I think we should avoid stack traces only for static sentinel Throwables.

@lachlan-roberts can you review your recent change in light of this constructor.

Reply:
I agree. While this code might be worth some refactorings, that's clearly outside the scope of the original issue which is just to avoid leaking stack traces in static exceptions.

I'm going to revert all these changes, introduce a new `StaticException` and keep the changes to a minimum.